orion/evaluation/contextual.py

```diff
@@ -0,0 +1,206 @@
+        data (DataFrame):
+            Original data, passed as a ``pandas.DataFrame`` containing timestamp.
+            Used to extract start and end.
+        start (int):
+            Minimum timestamp of the original data.
+        end (int):
+            Maximum timestamp of the original data.
+        method (function):
+            Function for computing confusion matrix (weighted segment, overlap segment).
+
+    Returns:
+        float:
+            Precision score between the ground truth and detected anomalies.
+    """
+    return _precision(expected, observed, data, start, end, method, contextual_confusion_matrix)
+
+
+def contextual_recall(expected, observed, data=None, start=None, end=None, method=None):
+    """Compute an recall score between the ground truth and the detected anomalies.
+
+    Args:
+        expected (DataFrame or list of tuples):
+            Ground truth passed as a ``pandas.DataFrame`` or list containing
+            two columns: start and stop.
+        observed (DataFrame or list of tuples):
+            Detected anomalies passed as a ``pandas.DataFrame`` or list containing
+            two columns: start and stop.
+        data (DataFrame):
+            Original data, passed as a ``pandas.DataFrame`` containing timestamp.
+            Used to extract start and end.
+        start (int):
+            Minimum timestamp of the original data.
+        end (int):
+            Maximum timestamp of the original data.
+        method (function):
+            Function for computing confusion matrix (weighted segment, overlap segment).
+
+    Returns:
+        float:
+            Recall score between the ground truth and detected anomalies.
+    """
+    return _recall(expected, observed, data, start, end, method, contextual_confusion_matrix)
+
+
+def contextual_f1_score(expected, observed, data=None, start=None, end=None, method=None):
+    """Compute an f1 score between the ground truth and the detected anomalies.
+
+    Args:
+        expected (DataFrame or list of tuples):
+            Ground truth passed as a ``pandas.DataFrame`` or list containing
+            two columns: start and stop.
+        observed (DataFrame or list of tuples):
+            Detected anomalies passed as a ``pandas.DataFrame`` or list containing
+            two columns: start and stop.
+        data (DataFrame):
+            Original data, passed as a ``pandas.DataFrame`` containing timestamp.
+            Used to extract start and end.
+        start (int):
+            Minimum timestamp of the original data.
+        end (int):
+            Maximum timestamp of the original data.
+        method (function):
+            Function for computing confusion matrix (weighted segment, overlap segment).
+
+    Returns:
+        float:
+            F1 score between the ground truth and detected anomalies.
+    """
+    return _f1_score(expected, observed, data, start, end, method, contextual_confusion_matrix)
+
+
+METRICS = {
```

Comment:
I would rename this to `CONTEXTUAL_METRICS`, to be consistent with the point ones.

Also, maybe these two dictionaries could be directly defined in the parent `__init__`?

Reply:
Updated.